src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +96,15 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                Clipboard.setContent(selection || SelectionScraper.getCustomSelection(messageHtml));
             } else {
-                Clipboard.setString(html);
+                Clipboard.setString(messageHtml);
```

Comment:
Here my initial suggestion was something like 
```js
   const message = _.last(lodashGet(reportAction, 'message', [{}]));
            const messageHtml = lodashGet(message, 'html', '');
            const isAttachment = _.has(reportAction, 'isAttachment')
                ? reportAction.isAttachment
                : ReportUtils.isReportMessageAttachment(message);
            if (!isAttachment) {
                try {
                    Clipboard.setContent(selection || messageHtml);
                } catch {
                    const parser = new ExpensiMark();
                    Clipboard.setContent({text: parser.htmlToMarkdown(selection || messageHtml)});
                }
            } else {
                Clipboard.setString(messageHtml);
            }
```            
It makes more sense to handle the fallback here if setContent failed to set it. What do you think?

Reply:
If we go with setHTML path, it becomes
```
  try {
                    Clipboard.setHTML(selection.html || messageHtml, {html: messageHTML, text: need to pass text});
                } catch {
                    const parser = new ExpensiMark();
                    Clipboard.setSrting(parser.htmlToMarkdown(selection || messageHtml));
                }